### PR TITLE
Document values.yaml fields.

### PR DESCRIPTION
**Pull request by bjhaid**
This documents all of the possible configuration of values.yaml with their descriptions, options and default values, all options that are required come early in the documentation while the rest follow.

### CODE REVIEW

File: configurator/configuration.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+sysdig.anchoreVersion | Version of the sysdig anchore | - | `v0.3.4.1` |
+sysdig.cassandraVersion | Version of Cassandra run by sysdig cloud | - | `2.1.21.13` |
+sysdig.elasticsearchVersion | Version of ElasticSearch run by sysdig cloud | - | `5.6.44.0` |
+sysdig.postgresVersion | Version of PostgreSQL run by sysdig cloud | - | `10.6.10` |
+sysdig.redisVersion | Version of Redis run by sysdig cloud | - | `4.0.12.5` |
+sysdig.redisHaVersion | Version of HA Redis run by sysdig cloud | - | `4.0.12.5-ha` |
+sysdig.haproxyVersion | Version of HAProxy run by sysdig cloud | - | `v0.7-beta.7` |
+sysdig.rsyslogVersion | Version of rsyslog run by sysdig cloud | - | `8.34.0.5` |
+sysdig.localVolumeProvisioner | Version of localVolumeProvisioner run by sysdig cloud | - | `v2.3.2` |
+elasticsearch.searchguard.enabled | - | `true\|false` | `true` |
```
Comment (bjhaid) [inline configurator/configuration.md]: @bob-bins I need you to help provide descriptions for the elasticsearch portions

File: configurator/configuration.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+sysdig.workerReplicaCount | Number of sysdig worker replicas, this is a noop for small clusters| - | small cluster: 1<br>medium cluster: 3<br>large cluster: 5|
+sysdig.redisHa | determines if redis should run in HA mode| `true\|false` | `false`|
+sysdig.smtpServer | specifies smtp server to use to send email| - | - |
+sysdig.smtpServerPort | specifies port for the configured smtp server| - | - |
+sysdig.smtpUser | specifies user for the configured smtp server| - | - |
+sysdig.smtpPassword | specifies password for the configured smtp user| - | - |
+sysdig.smtpProtocolTLS | specifies if TLS should be used for smtp| `true\|false` | - |
+sysdig.smtpProtocolSSL | specifies if SSL should be used for smtp| `true\|false` | - |
+sysdig.smtpFromAddress | specifies the email address for the FROM field of sent emails| - | - |
+sysdig.openshiftUrl | - | - | - |
```
Comment (bjhaid) [inline configurator/configuration.md]: @jpravin I need you to help provide description for the openshift pieces

Comment (jpravin) [reply configurator/configuration.md]: Added.

File: configurator/configuration.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
+Parameter | Description | Options | Default|
+|---------|-------------|------|--------|
+scripts | defines which scripts needs to be run<br> generate - performs templating & kustomization<br>deploy - applies the generated script in k8s environment| `generate\|deploy\|generate deploy`| - |
```
Comment (bob-bins) [inline configurator/configuration.md]: sort of a nitpick, but these would be easier to review here if the spacing better matched the column width, e.g.:
```
Parameter | Options                              | Default | Description 
----------|--------------------------------------|---------|-------------
scripts   | `generate | deploy | generate deploy`| -       | Defines which scripts needs to be run<br> generate - performs templating & kustomization<br>deploy - applies the generated script in k8s environment
apps      | `monitor | monitor secure`           | -       | Defines applications needed. Monitor installed by default.<br> Secure is optional. Remove secure if it is not deployed>
size      | `small | medium | large`             | -       | Size of the cluster.<br>This defines CPU & Memory & Disk & Replicas
```
Up to you whether it's worth the time to do this!

File: configurator/configuration.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+apps | defines applications needed. Monitor installed by default.<br> Secure is optional. Remove secure if it is not deployed> |`monitor\|monitor secure` | - |
```
Comment (bob-bins) [inline configurator/configuration.md]: does `secure` require `monitor`?

File: configurator/configuration.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+size | Size of the cluster.<br>This defines CPU & Memory & Disk & Replicas | `small\|medium\|large` | - |
```
Comment (bob-bins) [inline configurator/configuration.md]: all of these backslashes should be removed - they will render as literal backslashes

Comment (bob-bins) [reply configurator/configuration.md]: nevermind - my VS Code markdown plugin renders them as literal backslashes, but Github's renderer requires the backslashes

File: configurator/configuration.md (hunk 4 above)
Comment (bjhaid) [reply configurator/configuration.md]: yes it does require monitor

File: configurator/configuration.md (hunk 3 above)
Comment (bjhaid) [reply configurator/configuration.md]: it is going to have the same problem because of folds.

File: configurator/configuration.md (hunk 1 above)
Comment (bob-bins) [reply configurator/configuration.md]: Descriptions:
- `elasticsearch.searchguard.enabled`: Enables user authentication and TLS-encrypted data-in-transit with [Searchguard](https://search-guard.com/)

- `elasticsearch.searchguard.adminUser`: This isn't something the user should be able to change yet. It's tied to some other hardcoded values we have

- `elasticsearch.searchguard.verifySSL`: asdf

- `elasticsearch.jvmOptions`: asdf


```
elasticsearch.searchguard.enabled | Enables user authentication and TLS-encrypted data-in-transit with [Searchguard](https://search-guard.com/) | `true \| false` | `true` |
elasticsearch.searchguard.adminUser | The admin user that Searchguard uses | `admin` | `admin` |
elasticsearch.searchguard.verifySSL | Enables hostname verification when the backend components communicate with Elasticsearch nodes | `true\|false` | `false` |
elasticsearch.jvmOptions | - | - | "" |
```

**Comment by bjhaid** [comment]
This PR is ready for review please take a second pass

**Comment by bjhaid** [comment]
I am going to close this PR and open a new one, jenkins seems to be reporting failures I cannot reproduce, probably confused.

**Pull request closed without merging**